packages/list_bloc/lib/src/cubits/list_cubit.dart

```diff
@@ -20,5 +20,50 @@ class ListCubit<T, F> extends DataCubit<List<T>, F> {
       rethrow;
     }
   }
-// TODO(alexis): implement remove, add, replace etc
+
+  void add(T element) {
+    final data = <T>[...state.data ?? [], element];
+    emit(state.copyWith(data: data));
+  }
+
+  void remove(T element) {
+    final data = <T>[...state.data ?? []]..remove(element);
+    emit(
+      data.isEmpty
+          ? Data.empty(filter: state.filter)
+          : Data(data: data, filter: state.filter),
+    );
+  }
+
+  void removeWhere(bool Function(T) test) {
+    final data = <T>[...state.data ?? []]..removeWhere(test);
+    emit(
+      data.isEmpty
```

Comment:
It would be great to make the empty state check DRY as being used multiple times.